internals/plan/plan.go

```diff
@@ -42,11 +68,33 @@ const (
 	defaultCheckThreshold = 3
 )
 
+// layerExtensions keeps a map of registered extensions.
+var layerExtensions = map[string]LayerSectionExtension{}
```

Comment:
This feels bad. I think it's not that unrealistic to pass the extensions along the call chain from PlanManager to the Plan and into each Layer. Obviously this would be an invasive refactor.

Reply:
Yes, you are right this is a bit unusual.

I think the reason this was preferred was because the ```Plan``` library type is used as an immutable snapshot of the current plan, so its short lived. Throughout the lifetime of Pebble, new instances of ```Plan{}``` is created with the latest set of layers, and a combined plan. The extensions on the other hand stays the same throughout the lifetime of the Pebble process.

Instead of the Plan Manager receiving the extensions, and configuring the plan library once, as you say, the Plan Manager could ensure every ```Plan``` instance has a pointer to the extensions. I will keep this open and discuss the option with Gustavo / Ben.

Reply:
> Instead of the Plan Manager receiving the extensions, and configuring the plan library once, as you say, the Plan Manager could ensure every Plan instance has a pointer to the extensions.

Yeah this is more or less what I was thinking